plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -140,6 +145,8 @@ public PlanFilesIterable(CloseableIterable<FileScanTask> planFiles, Map<Integer,
             this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
             this.integerToBigintMapType = typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()));
             this.integerToVarcharMapType = typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()));
+            this.integerArrayType = typeManager.getType(arrayType(INTEGER.getTypeSignature()));
+            this.bigintArrayType = typeManager.getType(arrayType(INTEGER.getTypeSignature()));
```

Comment:
`BIGINT`?

Reply:
nit: I'd have declared and assigned in order (same as order of `tableMetadata`) of `bigintArrayType` and then `integerArrayType`.

Reply:
Nice catch. Let me update the PR.